### DISCUSSION TITLE
how to use PLAIN authentication to connect the kafka cluster in confluent.cloud ?

### DISCUSSION THREAD

**Discussion by xiaoronglv** [zendesk/ruby-kafka#908] [Q&A] [answered]
Hi there,

I have created a kafka cluster in confluent.cloud. They provide the endpoint, username, and password.

```
bootstrap.servers=pkc-xxxx.us-east-2.aws.confluent.cloud:9092
security.protocol=SASL_SSL
sasl.mechanisms=PLAIN
sasl.username=***
sasl.password=***
```

when using `ruby-kafka` to connect the remote kafka, an error has been raised.

```
kafka = Kafka.new(
  ["pkc-xxxxx.us-east-2.aws.confluent.cloud:9092"],
  sasl_plain_username: 'xxxxxxx',
  sasl_plain_password: 'xxxxxxxx'
)
```

error message: `ArgumentError: SASL authentication requires that SSL is configured`. 

what does that mean?  How can I make it work? should I specify a public key?

**Comment by xiaoronglv** [comment]
The example code of Plain authentication likes this.

```
kafka = Kafka.new(
  ["kafka1:9092"],
  ssl_ca_cert: File.read('/etc/openssl/cert.pem'),
  sasl_plain_username: 'username',
  sasl_plain_password: 'password'
  # ...
)
```

But I have no idea how to get the pem file.

**Comment by xiaoronglv** [comment, marked as answer]
update: I have checked their source code of cli tool. the solution is quiet simple.

https://github.com/confluentinc/examples/blob/66d7507d39fd64031a49cec97dac82a45a895c61/clients/cloud/ruby/c_cloud.rb#L41

```
kafka = Kafka.new(
  ["kafka1:9092"],
  ssl_ca_certs_from_system: true, # this is the key line.
  sasl_plain_username: 'username',
  sasl_plain_password: 'password'
  # ...
)

**Comment by avinashr0811** [comment]
use `sasl_over_ssl: false`
